src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/ApplicationService.kt

```diff
@@ -251,6 +251,54 @@ class ApplicationService(
     return success(createdApplication.apply { schemaUpToDate = true })
   }
 
+  fun createCas2Application(crn: String, user: UserEntity, jwt: String) = validated<ApplicationEntity> {
+    val offenderDetailsResult = offenderService.getOffenderByCrn(crn, user.deliusUsername)
+
+    val offenderDetails = when (offenderDetailsResult) {
+      is AuthorisableActionResult.NotFound -> return "$.crn" hasSingleValidationError "doesNotExist"
+      is AuthorisableActionResult.Unauthorised -> return "$.crn" hasSingleValidationError "userPermission"
+      is AuthorisableActionResult.Success -> offenderDetailsResult.entity
+    }
+
+    if (offenderDetails.otherIds.nomsNumber == null) {
+      throw RuntimeException("Cannot create an Application for an Offender without a NOMS number")
+    }
+
+    if (validationErrors.any()) {
+      return fieldValidationError
+    }
+
+    var riskRatings: PersonRisks? = null
+
+    val riskRatingsResult = offenderService.getRiskByCrn(crn, jwt, user.deliusUsername)
+
+    riskRatings = when (riskRatingsResult) {
+      is AuthorisableActionResult.NotFound -> return "$.crn" hasSingleValidationError "doesNotExist"
+      is AuthorisableActionResult.Unauthorised -> return "$.crn" hasSingleValidationError "userPermission"
+      is AuthorisableActionResult.Success -> riskRatingsResult.entity
+    }
+
+    val createdApplication = applicationRepository.save(
+      Cas2ApplicationEntity(
+        id = UUID.randomUUID(),
+        crn = crn,
+        createdByUser = user,
+        data = null,
+        document = null,
+        // TODO: replace this with a CAS-2 specific JSON schema
```

Comment:
I'll leave a more detailed general comment about the JSON Schema stuff but just a note here that you should do this now rather than later